### PR TITLE
Fix --controls option was not working for dependent profile

### LINKED ISSUE #5647: --controls does not support dependent profiles
I tried the --controls and --tags features on dependent profiles, but they don't seem to work for me.

#5596 , @Nik08 
#5434 , @Vasu1105 

Thanks!

### PR DESCRIPTION
**Pull request by Vasu1105**
Signed-off-by: Vasu1105 <vasundhara.jagdale@chef.io>

## Description

- This fix the filtering not working for the dependent profile with `--controls` options
- Tested the same for `include_controls` and `require_controls`
- Tested the profile_a > profile_b > profile_c > profile_d >profile_e
## Related Issue

Fix #5647 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by aaronlippold** [comment]
Some good testing content is here:

https://saf.cms.gov/#/validate

this one is three levels of chain:

https://github.com/CMSgov/cms-ars-3.1-moderate-aws-rds-oracle-database-12c-stig-overlay

**Comment by aaronlippold** [comment]
I woiuld suggest the unit test to to at least 4 to 5 levels of dep chain:

CORE -> ORG -> CLOUD -> TEAM -> REGION 

Is now a common pattern

File: lib/inspec/dsl.rb [hunk 1]
```diff
@@ -93,21 +93,27 @@ def self.load_spec_files_for_profile(bind_context, opts, &block)
...
     bind_context.add_subcontext(context)
   end
 
-  def self.filter_included_controls(context, profile, &block)
+  def self.filter_included_controls(context, profile, opts, &block)
     mock = Inspec::Backend.create(Inspec::Config.mock)
     include_ctx = Inspec::ProfileContext.for_profile(profile, mock)
     include_ctx.load(block) if block_given?
+    # this sets the conf variable required in control_exist_in_control_list? method
+    include_ctx.control_eval_context.instance_variable_set(:@conf, opts[:conf])
```
Comment (clintoncwolfe) [inline lib/inspec/dsl.rb]: We should almost never resort to calling instance_variable_set on our own code. Can't we make conf an attr_accessor on ControlEvalContext and then just set it properly?

**Review by clintoncwolfe — COMMENTED** [review]
One quibble about how an instance variable is getting set, but otherwise this looks good to me and is well-tested.  Thanks!

**Comment by aaronlippold** [comment]
any blockers on this?

**Comment by clintoncwolfe** [comment]
@Vasu1105 , could you take care of the merge conflicts? Then we can get this merged in.

**Comment by Vasu1105** [comment]
> @Vasu1105 , could you take care of the merge conflicts? Then we can get this merged in.

Conflicts resolved. Thanks, @clintoncwolfe.

**Review by clintoncwolfe — APPROVED** [review]
Thanks!

**Comment by ccoffin** [comment]
The functionality seems to be working when specifying controls on the cli using `--controls` option only loading the controls i specify. However, when using a profile dependency and `require_controls` i specify it still seems to be loading all of the controls in the parent profile. Tested with both Inspec version 4.46.13 and cinc-auditor 4.46.13 on RHEL 8.

**Comment by Vasu1105** [comment]
@ccoffin could you please open a new issue with your use case in detail so that we can try reproducing it and address it.

**Comment by ccoffin** [comment]
@Vasu1105 Sorry for the delay. I had to rebuild my environment before typing up the issue. :-(

I created an issue for this, see #5705.

**Pull request merged**